File: Webpage/server/webpage.server/webpage.server/src/main/java/abbruzzese/webpage/webpage/server/webpage/entities/Product.java
package abbruzzese.webpage.webpage.server.webpage.entities;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import java.time.LocalDate;
import java.util.UUID;

@Entity
public class Product {

    public Product (String Nombre, String Marca, String categoria,LocalDate fecha, String imageproduct,String enlace){
   
        this.nombre=Nombre;
        this.Marca=Marca;
        this.Categoria=categoria;
        this.fecha=fecha;
        this.imageproduct=imageproduct;

    }

    public Product(){}

    @Id
    private String id=UUID.randomUUID().toString();;

    @Column(name="Nombre",unique = true,nullable=false)
    private String nombre;

    @Column (name="Marca",nullable=false)
    private String Marca;

    @Column(name="Categoria",nullable=false)
    private String Categoria;

    @Column(name="Fecha",nullable=false)
    private LocalDate fecha;

    @PrePersist
    public void prePersist() {
        this.fecha = LocalDate.now(); // Asignar la fecha actual al campo createdAt
    }

    @Column(name = "ImageProduct",nullable = false)
    private String imageproduct;

    public String getIduser() {
        return id;
    }

    public void setIduser(String id) {
        this.id = id;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String Nombre) {
        nombre = Nombre;
    }

    public String getMarca() {
        return Marca;
    }

    public void setMarca(String marca) {
        Marca = marca;
    }

    public String getCategoria() {
        return Categoria;
    }

    public void setCategoria(String categoria) {
        Categoria = categoria;
    }
       
    public LocalDate getFecha() {
        return fecha;
    }

    public void setFecha(LocalDate fecha) {
        this.fecha = fecha;
    }

    public String getImageproduct() {
        return imageproduct;
    }

    public void setImageproduct(String imageproduct) {
        this.imageproduct = imageproduct;
    }

}

File: PHP_API/ENTITIES/Product.php
<?php 

namespace App\Entity;

use Ramsey\Uuid\Uuid;
use Doctrine\ORM\Mapping as ORM;
use App\Repository\ProductRepository;

#[ORM\Entity(repositoryClass: ProductRepository::class)] // Esto usa atributos de PHP 8
#[ORM\Table(name: "product")]
class Product{
    
    #[ORM\Id]
    #[ORM\Column(type: "string")]
    #[ORM\GeneratedValue(strategy:"NONE")]
    private $id;

    #[ORM\Column(type:"string",unique: true)]
    private $nombre;

    #[ORM\Column(type:"string")]
    private $marca;

    #[ORM\Column(type:"string",unique: true)]
    private $categoria;

    #[ORM\Column(type:"date")]
    private $localdate;

    #[ORM\Column(type:"string")]
    private $imageproduct;

    public function __construct($nombre,$marca,$categoria,$imageproduct) {
        $this->id = Uuid::uuid4();
        $this->nombre = $nombre;
        $this->marca = $marca;
        $this->categoria = $categoria;
        $this->localdate = \DateTime::createFromFormat('Y-m-d', date('Y-m-d'));; // Establece la fecha y hora actual
        $this->imageproduct=$imageproduct;
    }


    // Getters y setters
    public function getId() {return $this->id;}

    public function getNombre() {return $this->nombre;}
    public function setNombre($nombre){ $this->nombre=$nombre;}

    public function getMarca() {return $this->marca;}
    public function setMarca($marca){$this->marca=$marca;}

    public function getCategoria() {return $this->categoria;}
    public function setCategoria($categoria){$this->categoria=$categoria;}

    public function getLocaldate() {return $this->localdate;}
    public function setLocaldate($localdate){$this->localdate=$localdate;}

    public function getImageproduct() {return $this->imageproduct;}
    public function setImageproduct($imageproduct){$this->imageproduct=$imageproduct;}

}

?>